IndexOutOfBoundsException in sitesearch portlet

### Parent Issue

_No response_

### Problem Statement

A customer reported an issue related to sitesearch and in the logs shows.

``` ERROR servlet.VisitorFilter: An exception occurred processing [/html/portlet/ext/sitesearch/site_search_index_stats.jsp] at line [43]

40: 
41: 
42: 
43: List<String> indices=ssapi.listIndices();
44: List<String> closedIndices=ssapi.listClosedIndices();
45: 
46: Map<String, IndexStats> indexInfo = esapi.getIndicesStats();


Stacktrace:
org.apache.jasper.JasperException: An exception occurred processing [/html/portlet/ext/sitesearch/site_search_index_stats.jsp] at line [43]

40: 
41: 
42: 
43: List<String> indices=ssapi.listIndices();
44: List<String> closedIndices=ssapi.listClosedIndices();
45: 
46: Map<String, IndexStats> indexInfo = esapi.getIndicesStats();


Stacktrace:
at org.apache.jasper.servlet.JspServletWrapper.handleJspException(JspServletWrapper.java:610) ~[jasper.jar:9.0.60]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:499) ~[jasper.jar:9.0.60]
at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:379) ~[jasper.jar:9.0.60]
at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:327) ~[jasper.jar:9.0.60]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) ~[servlet-api.jar:4.0.FR]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-websocket.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotmarketing.filters.CMSFilter.doFilterInternal(CMSFilter.java:181) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at com.dotmarketing.filters.CMSFilter.doFilter(CMSFilter.java:61) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotcms.visitor.filter.servlet.VisitorFilter.doFilter(VisitorFilter.java:81) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotcms.vanityurl.filters.VanityURLFilter.doFilter(VanityURLFilter.java:114) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[urlrewritefilter-4.0.4.jar:4.0.4]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotmarketing.filters.TimeMachineFilter.doFilter(TimeMachineFilter.java:137) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotmarketing.filters.ThreadNameFilter.doFilter(ThreadNameFilter.java:88) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotmarketing.filters.CookiesFilter.doFilter(CookiesFilter.java:53) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotmarketing.filters.CharsetEncodingFilter.doFilter(CharsetEncodingFilter.java:99) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotcms.filters.interceptor.AbstractWebInterceptorSupportFilter.doFilter(AbstractWebInterceptorSupportFilter.java:90) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:126) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at com.dotcms.filters.NormalizationFilter.doFilter(NormalizationFilter.java:89) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[catalina.jar:9.0.60]
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) ~[catalina.jar:9.0.60]
at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687) ~[catalina.jar:9.0.60]
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[catalina.jar:9.0.60]
at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:769) ~[catalina.jar:9.0.60]
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[catalina.jar:9.0.60]
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360) ~[catalina.jar:9.0.60]
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399) ~[tomcat-coyote.jar:9.0.60]
at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-coyote.jar:9.0.60]
at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:889) ~[tomcat-coyote.jar:9.0.60]
at org.apache.tomcat.util.net.Nio2Endpoint$SocketProcessor.doRun(Nio2Endpoint.java:1701) ~[tomcat-coyote.jar:9.0.60]
at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:9.0.60]
at org.apache.tomcat.util.net.AbstractEndpoint.processSocket(AbstractEndpoint.java:1190) ~[tomcat-coyote.jar:9.0.60]
at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$2.completed(Nio2Endpoint.java:633) ~[tomcat-coyote.jar:9.0.60]
at org.apache.tomcat.util.net.Nio2Endpoint$Nio2SocketWrapper$2.completed(Nio2Endpoint.java:611) ~[tomcat-coyote.jar:9.0.60]
at sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:127) ~[?:?]
at sun.nio.ch.UnixAsynchronousSocketChannelImpl.finishRead(UnixAsynchronousSocketChannelImpl.java:439) ~[?:?]
at sun.nio.ch.UnixAsynchronousSocketChannelImpl.finish(UnixAsynchronousSocketChannelImpl.java:191) ~[?:?]
at sun.nio.ch.UnixAsynchronousSocketChannelImpl.onEvent(UnixAsynchronousSocketChannelImpl.java:213) ~[?:?]
at sun.nio.ch.EPollPort$EventHandlerTask.run(EPollPort.java:306) ~[?:?]
at sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:112) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 3
at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248) ~[?:?]
at java.util.Objects.checkIndex(Objects.java:372) ~[?:?]
at java.util.ArrayList.remove(ArrayList.java:536) ~[?:?]
at com.dotcms.enterprise.publishing.sitesearch.ESSiteSearchAPI.setDefaultToSpecificPosition(ESSiteSearchAPI.java:168) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at com.dotcms.enterprise.publishing.sitesearch.ESSiteSearchAPI.listIndices(ESSiteSearchAPI.java:150) ~[dotcms_23.10.24_lts_v2_5374aed.jar:?]
at org.apache.jsp.html.portlet.ext.sitesearch.site_005fsearch_005findex_005fstats_jsp._jspService(site_005fsearch_005findex_005fstats_jsp.java:553) ~[?:?]
at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70) ~[jasper.jar:9.0.60]
at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) ~[servlet-api.jar:4.0.FR]
at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:466) ~[jasper.jar:9.0.60]
... 77 more
```


### Steps to Reproduce

1. Create a few SS indices
2. Set one as default
3. Delete the default SS index directly from ES:
```
curl -X DELETE "http://localhost:9200/{index_name}"
```
4. Flush Cache
5. Try to create a new SS index

This line throws an error  
https://github.com/dotCMS/core/blob/8b71ba0ef58e824ad5e63645e65c1285bcaa6847/dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/sitesearch/ESSiteSearchAPI.java#L163
 
Because list.indexOf(defaultIndice); returns -1 if default indices are not there in the list
 
```
final int index = list.indexOf(defaultIndice);
                    //change the element defaultIndex to the first position of the arraylist if it is not yet
                    if (index != 0) {
                        list.remove(index);
                        list.add(indexPosition, defaultIndice);
                    }
```
 
The root cause is here. 
indiciesAPI.loadIndicies() returns different results
 https://github.com/dotCMS/core/blob/8b71ba0ef58e824ad5e63645e65c1285bcaa6847/dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/sitesearch/ESSiteSearchAPI.java#L161
 

### Acceptance Criteria

The logic error should handle and indexoutofbound should never return if difference in indeces

### dotCMS Version

24.*

### Proposed Objective

Customer Support

### Proposed Priority

Priority 4 - Trivial

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/116403

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/sitesearch/ESSiteSearchAPI.java
+++ b/dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/sitesearch/ESSiteSearchAPI.java
@@ -162,7 +162,9 @@ public class ESSiteSearchAPI implements SiteSearchAPI{
                 if (defaultIndice != null && !defaultIndice.isEmpty() && !list.isEmpty() ){
                     final int index = list.indexOf(defaultIndice);
                     //change the element defaultIndex to the first position of the arraylist if it is not yet
-                    if (index != 0) {
+                    if(index < 0){
+                        Logger.warn(this.getClass(), String.format("The default site search '%s' index was not found in the list of indices.", defaultIndice));
+                    } else {
                         list.remove(index);
                         list.add(indexPosition, defaultIndice);
                     }

--- a/dotcms-integration/src/test/java/com/dotcms/content/elasticsearch/business/ESSiteSearchAPITest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/content/elasticsearch/business/ESSiteSearchAPITest.java
@@ -194,6 +194,38 @@ public class ESSiteSearchAPITest {
         assertEquals(defIndex, indices.get(0));
     }
 
+    /**
+     * Method to test: {@link SiteSearchAPI#listIndices()}
+     * Given Scenario: Create a few SiteSearch, set one as default, delete it, Attempt to load the list.
+     * ExpectedResult: The list should load fine, a WARN message should be logged regarding Default Index not found.
+     */
+    @Test
+    public void test_listIndices_defaultIndexNotExist() throws IOException, DotDataException {
+        String timeStamp, indexName, aliasName;
+        String defIndex = "";
+
+        final int indicesAmount = 3;
+        for (int i = 0; i < indicesAmount; i++) {
+            timeStamp = String.valueOf(new Date().getTime());
+            indexName = ES_SITE_SEARCH_NAME + "_" + timeStamp;
+            aliasName = "indexAlias" + "_" + timeStamp;
+
+            siteSearchAPI.createSiteSearchIndex(indexName, aliasName, 1);
+
+            if (i == 2){
+                //sets as default
+                siteSearchAPI.activateIndex(indexName);
+                defIndex = indexName;
+            }
+        }
+
+        //delete the default index
+        indexAPI.delete(defIndex);
+
+        //get the list of indices (previously was throwing an IndexOutOfBoundsException)
+        siteSearchAPI.listIndices();
+    }
+
 
     /**
      * Method to test: {@link SiteSearchAPI#deleteOldSiteSearchIndices()}

